Added option to do WTA reclustering for jet axis

jusaviin opened the pull request:
Added option to do WTA reclustering for the jet axis also for the 2018 data processing branch. The changes are otherwise the same as in my previous commit to CMSSW_7_5_8_patch3, but it seems that I do not have to disable calo jets for tests to work in the newer branch. So as all the tests work even without disabling calo jets, I also include those for WTA axis reclustering if the option is enabled.

Commit:
Added option to do WTA reclustering for jet axis

bi-ran commented on HeavyIonsAnalysis/JetAnalysis/src/HiInclusiveJetAnalyzer.cc:
```diff
@@ -691,6 +700,12 @@ HiInclusiveJetAnalyzer::beginJob() {
 
+      //for reWTA reclustering
+      if(doWTARecluster_){
+        t->Branch("WTAgeneta",jets_.WTAgeneta,"WTAgeneta[nref]/F");
```
@jusaviin @FHead should this be ngen? if yes, please also fix this in 758p3

jusaviin replied:
Yes, it should be ngen. Thank you for noticing. I will correct that here and in 758p3.

jusaviin committed:
Fixed array sizes for gen WTA axis

The generator level WTA clustering had accidentally eta and phi array sizes coming from the number of reco jets (nref). The correct size is given by the number of gen jets (ngen).

bi-ran commented:
thanks! @jusaviin
